src/main/java/io/github/oliviercailloux/j_voting/preferences/interfaces/CompletePreference.java

```diff
@@ -0,0 +1,48 @@
+package io.github.oliviercailloux.j_voting.preferences.interfaces;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import io.github.oliviercailloux.j_voting.Alternative;
+
+/**
+ * A Complete Preference is is an immutable preference. A complete preference
+ * represents a complete pre-order, also called a weak order. for each couple of
+ * alternatives (a,b) we can find an order a>=b or b>=a.
+ */
+public interface CompletePreference extends ImmutablePreference {
+
+    /**
+     *
+     * @param a is an <code>Alternative</code>
+     * @return the rank of this alternative (a number between 1 and <i>n</i>
+     * where <i>n</i> is the total number of <code>Alternative</code><br><br>
```

Comment:
Enlever les `<br>`, la mise en page est le boulot de l’outil de génération de doc. Fermer la parenthèse. Écrire « \<code>Alternative\</code> instances » ou plus simplement « alternatives ».

Il faut aussi préciser comment les rangs sont comptés car il y a ambiguïté en cas d’ex-æquo. Écrire par exemple : the rank of an alternative is one plus the number of alternatives strictly preferred to it.